src/components/text-editor/link-menu/limel-text-editor-link-menu.tsx

```diff
@@ -0,0 +1,150 @@
+import { Component, Prop, h, Event, EventEmitter, State } from '@stencil/core';
+import { EditorTextLink } from '../prosemirror-adapter/menu/types';
+import { Languages } from '../../date-picker/date.types';
+import translate from '../../../global/translations';
+
+/**
+ * This component is a menu for editing a link in the text editor.
+ * It allows the user to input a title of a link and a link itself.
+ */
+
+@Component({
+    tag: 'limel-text-editor-link-menu',
+    shadow: true,
+    styleUrl: 'limel-text-editor-link-menu.scss',
+})
+export class TextEditorLinkMenu {
+    /**
+     * The link
+     */
+    @Prop()
+    public link: EditorTextLink;
+
+    /**
+     * Defines the language for translations.
+     * Will translate the translatable strings on the components.
+     */
+    @Prop({ reflect: true })
+    public language: Languages = 'en';
+
+    /**
+     * Open state of the dialog
+     */
+    @Prop()
+    public isOpen: boolean = false;
+
+    @State()
+    private invalidLink: boolean = false;
+
+    /**
+     * Emitted when the menu is closed from inside the component.
+     * (*Not* emitted when the consumer sets the `open`-property to `false`.)
+     */
+    @Event()
+    private cancel: EventEmitter<void>;
+
+    /**
+     * Emitted when the menu is saved from inside the component.
+     */
+    @Event()
+    private save: EventEmitter<void>;
+
+    /**
+     * Emitted when the user inputs new values for the link
+     */
+    @Event()
+    private linkChange: EventEmitter<EditorTextLink>;
+
+    public render() {
+        return [
+            <limel-input-field
+                label="Text"
+                value={this.link?.text || ''}
+                leadingIcon="text_cursor"
+                onChange={this.handleLinkTitleChange}
+            />,
+            <limel-input-field
+                label="Link"
+                value={this.link?.href || ''}
+                leadingIcon="-lime-text-link"
+                invalid={this.invalidLink}
+                onChange={this.handleLinkValueChange}
+            />,
+            <div class="actions">
+                <limel-button
+                    label={this.getTranslation('cancel')}
+                    onClick={this.handleCancel}
+                    slot="button"
+                />
+                <limel-button
+                    primary={true}
+                    label={this.getTranslation('save')}
+                    disabled={!this.link?.href || this.invalidLink}
+                    onClick={this.handleSave}
+                    slot="button"
+                />
+            </div>,
+        ];
+    }
+
+    private getTranslation(key: string) {
+        return translate.get(`${key}`, this.language);
```

Comment:
Correct me if I'm missing something, but…

```suggestion
        return translate.get(key, this.language);
```

?

🙂

Reply:
I have no idea on this one. It most certainly has no need for a string literal.